src/main/kotlin/WorkerResult.kt

```diff
@@ -1,6 +1,23 @@
 package franz
 
-enum class WorkerResult {
+class WorkerResult<out L> (
+    val value: L? = null,
+    val status: WorkerStatus
+) {
+    companion object {
+        fun <L> success(result: L) = WorkerResult(result, WorkerStatus.Success)
+        val retry = WorkerResult(null, WorkerStatus.Retry)
+        val failure = WorkerResult(null, WorkerStatus.Failure)
+    }
+
+    fun toJobStatus(): JobStatus = when(status){
+        WorkerStatus.Success -> JobStatus.Incomplete
```

Comment:
Is this correct?

Reply:
Yes. Incomplete means the pipe continues. A jobstate with the status Success will be considered succesfull and therefor not executing more steps.